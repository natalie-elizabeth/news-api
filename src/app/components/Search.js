import Search from 'react-search'
import ReactDOM from 'react-dom'
import React, { Component, PropTypes } from 'react'

export default class Searching extends Component {

    constructor(props) {
        super(props)
        this.state = { repos: [] }
    }

    getItemsAsync(searchValue, cb) {
        let url = `https://api.github.com/search/repositories?q=${searchValue}&language=javascript`
        fetch(url).then((response) => {
            return response.json();
        }).then((results) => {
            if (results.items != undefined) {
                let items = results.items.map((res, i) => { return { id: i, value: res.full_name } })
                this.setState({ repos: items })
                cb(searchValue)
            }
        });
    }

    render() {
        return (
            <div>
                <Search items={this.state.repos}
                    multiple={true}
                    getItemsAsync={this.getItemsAsync.bind(this)}
                    onItemsChanged={this.HiItems.bind(this)} />
            </div>
        )
    }
}