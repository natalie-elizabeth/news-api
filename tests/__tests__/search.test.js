import { shallow, render, mount } from 'enzyme';
import React from 'react';
import Search from '../../src/app/components/Search';
import shallowToJSON from 'enzyme-to-json';
import ReactTestUtils from 'react-dom/test-utils';

describe('Search', () => {
  it('check for presence of div elements', () => {
    const wrapper = shallow(<div></div>);
    expect(wrapper.find('div').node).toBeDefined();
  });
  it('should have a search button', () => {
    const wrapper = shallow(<Search />);
    const tree = shallowToJSON(wrapper);
    expect(tree).toMatchSnapshot();
  });
  //   it ('should display text in the button', () => {
  //     const wrapper = shallow(<)
  // const text = 'Search';
});



//   // it('should display search button', () => {
//   //     const renderedComponent = shallow(<searchbox />);
//   //     expect(renderedComponent.find("search").node).toBeDefined();
//   // });

//   it('should display search text in the button', () => {
//     const text = 'Search';
//     const renderedComponent = shallow(<div>{'Search'}</div>);
//     expect(renderedComponent.contains(text)).toBe(true);
//   });

//   it("trigger onSearch correctly", () => {
//     const handleSearch = jest.fn();
//   });
// });
