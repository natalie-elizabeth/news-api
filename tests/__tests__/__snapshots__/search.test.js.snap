// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should have a search button 1`] = `
<div
  className="col-md-2"
  id="search"
>
  <input
    className="searchbox"
    onChange={[Function]}
    placeholder="Source Name"
    style={
      Object {
        "color": "grey",
      }
    }
    type="text"
    value="Search"
  />
</div>
`;
